pkg/daemon/ceph/osd/remove.go

```diff
@@ -140,13 +140,6 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 	// call archiveCrash to silence crash warning in ceph health if any
 	archiveCrash(clusterdContext, clusterInfo, osdID)
 
-	// remove the auth for the osd. should be done by 'osd purge', but doesn't always seem to happen
```

Comment:
Why not just leave this in case there is a conflict from not stopping the operator? Not many people read the docs, after all.

Reply:
It's a good thought, but this doesn't actually fix the conflict. The race condition will still have created a new OSD deployment, and Rook will re-create the deployment with every new reconcile until the disk is wiped/removed.